core/domain/exp_domain.py

```diff
@@ -2127,16 +2134,29 @@ def extract_content_id_from_choices(html: str) -> str:
                 return feconf.INVALID_CONTENT_ID
 
             if new_type == 'TranslatableHtmlContentId':
+                # Here 'TranslatableHtmlContentId' can only be of str type, thus
+                # to narrow down the type we used assert here.
+                assert isinstance(value, str)
                 return extract_content_id_from_choices(value)
             elif new_type == 'SetOfTranslatableHtmlContentIds':
+                # Here 'migrate_rule_inputs_and_answers' method calling itself
+                # recursively and because of this MyPy assuming it's type as
+                # recursive, like if this method returns List[str] then MyPy
+                # assumes it's type as List[List[str]]. So because of this MyPy
```

Comment:
So, because of this, ...


(Similar comments to these three apply below as well.)

Reply:
changed